.github/workflows/spread-tests.yaml

```
@@ -0,0 +1,274 @@
+        path: "${{ github.workspace }}/built-snap"
+
+    - name: Download built FIPS snap
+      uses: actions/download-artifact@v4
+      # eg. ubuntu-fips
+      if: "endsWith(inputs.group, '-fips')"
+      with:
+        name: snap-files-FIPS-test
+        # eg. snapd_1337.2.65.1+git97.gd35b459-fips_amd64.snap
+        pattern: snapd_1337.*-fips_*.snap
+        path: "${{ github.workspace }}/built-snap"
+
+    - name: Rename imported snap
+      run: |
+        for snap in built-snap/snapd_1337.*.snap; do
+          mv -v "${snap}" "${snap}.keep"
+        done
+
+    - name: Run spread tests
+      if: "!contains(github.event.pull_request.labels.*.name, 'Skip spread') && ( !startsWith(inputs.group, 'nested-') || contains(github.event.pull_request.labels.*.name, 'Run nested') )"
+      env:
+          SPREAD_GOOGLE_KEY: ${{ secrets.SPREAD_GOOGLE_KEY }}
+      run: |
+          # Register a problem matcher to highlight spread failures
+          echo "::add-matcher::.github/spread-problem-matcher.json"
+          set -x
+          SPREAD=spread
+          if [[ "${{ inputs.group }}" =~ nested- ]]; then
+            export NESTED_BUILD_SNAPD_FROM_CURRENT=true
+            export NESTED_ENABLE_KVM=true
+          fi
+
+          case "${{ inputs.systems }}" in
+            *-arm-*)
+              SPREAD_USE_PREBUILT_SNAPD_SNAP=false
+              ;;
+            *)
+              SPREAD_USE_PREBUILT_SNAPD_SNAP=true
+              ;;
+          esac
+          export SPREAD_USE_PREBUILT_SNAPD_SNAP
+
+          if [[ "${{ inputs.systems }}" =~ amazon-linux-2023 ]]; then
+              # Amazon Linux 2023 has no xdelta, however we cannot disable
+              # xdelta on a per-target basis as it's used in the repack section
+              # of spread.yaml, which is shared by all targets, so all systems
+              # in this batch will not use delta for transferring project data
+              echo "Disabling xdelta support"
+              export NO_DELTA=1
+          fi
+
+          # Add openstack backend definition to spread.yaml
+          if [ "${{ inputs.backend }}" = openstack ]; then
+              ./tests/lib/spread/add-backend tests/lib/spread/backend.openstack.yaml spread.yaml
+          fi
+
+          # This coud be the case when either there are not systems for a group or 
+          # the list of tests to run is empty
+          if [ -z "$RUN_TESTS" ]; then
+            echo "No tests to run, skiping..."
+            exit 0
+          fi
+
+          # Run spread tests
+          # "pipefail" ensures that a non-zero status from the spread is
+          # propagated; and we use a subshell as this option could trigger
+          # undesired changes elsewhere
+          echo "Running command: $SPREAD $RUN_TESTS"
+          (set -o pipefail; $SPREAD -no-debug-output -logs spread-logs $RUN_TESTS | PYTHONDONTWRITEBYTECODE=1 ./tests/lib/external/snapd-testing-tools/utils/log-filter $FILTER_PARAMS | tee spread.log)
```

since we're already modifying the code:
```suggestion
          (
              set -o pipefail
              $SPREAD -no-debug-output -logs spread-logs $RUN_TESTS | \
                  PYTHONDONTWRITEBYTECODE=1 ./tests/lib/external/snapd-testing-tools/utils/log-filter $FILTER_PARAMS | \
                  tee spread.log
          )
```

---

FWIW, not need to do this now, a followup PR is fine.

---

Thanks, I'll add this to the clean-up PR to follow